docs/docs/workflows/steps/code/async/README.md

```diff
@@ -0,0 +1,116 @@
+# Running asynchronous code
+
+If you're not familiar with asynchronous programming concepts like [callback functions](https://developer.mozilla.org/en-US/docs/Glossary/Callback_function) or [Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises), see [this overview](https://eloquentjavascript.net/11_async.html).
+
+[[toc]]
+
+## The problem
+
+**Any asynchronous code within a Node.js code step must complete before the next step runs**. This ensures future steps have access to its data. If Pipedream detects that code is still running by the time the step completes, you'll see the following warning below the code step:
+
+> **This step was still trying to run code when the step ended. Make sure you await all Promises, or promisify callback functions.**
+
+As the warning notes, this often arises from one of two issues:
+
+- You forgot to `await` a Promise. [All Promises must be awaited](#await-all-promises) so they run synchronously.
```

Comment:
In both these cases I would say it's not about them running synchronously (because they are still async) but more about just ensuring the wrapping step function properly awaits their resolution.